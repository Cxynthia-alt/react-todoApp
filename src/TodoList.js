import React, { useState } from 'react';
import Todo from './Todo'
import { v4 as uuid } from 'uuid'
import NewTodoForm from './NewTodoForm'

const TodoList = () => {
  // const INITIAL_STATE = [
  //   { id: uuid(), item: "laundry" },
  //   { id: uuid(), item: "shopping" }
  // ]
  const INITIAL_STATE = []
  const [todoList, setTodoList] = useState(INITIAL_STATE)
  const addTodo = (newTodo) => {
    setTodoList(todolist => [
      ...todoList,
      { item: newTodo.item, id: uuid() }
    ])
  }

  return (
    <div>
      <h3>TodoList</h3>
      <NewTodoForm addTodo={addTodo} />
      {todoList.map(todo => <Todo item={todo.item} key={todo.id} todoList={todoList} setTodoList={setTodoList} id={todo.id} />)}
    </div>
  )
}

export default TodoList
