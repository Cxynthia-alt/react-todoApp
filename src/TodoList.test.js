import React from "react";
import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList"


it("render without crashing", () => {
  render(<TodoList />)
})

it("matches snapshot", () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
})

it("can add/delete a todo item", () => {
  const todoList = render(<TodoList />)

  //add todo
  const itemInput = todoList.getByLabelText("New item")
  const addBtn = todoList.getByText("Add Me!");
  fireEvent.change(itemInput, { target: { value: "cleaning" } })
  fireEvent.click(addBtn)

  //expect to see a todo
  expect(todoList.getByLabelText("New item")).toHaveValue("");
  expect(todoList.getByText("X")).toBeInTheDocument();

  //expect to delete a todo
  const deleteBtn = todoList.getByText("X");
  fireEvent.click(deleteBtn)
  expect(todoList.getByLabelText("New item")).toHaveValue("");

})
