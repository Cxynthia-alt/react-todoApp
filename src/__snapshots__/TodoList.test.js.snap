// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      TodoList
    </h3>
    <div>
      <form>
        <label
          for="item"
        >
          New item
        </label>
        <input
          id="item"
          name="item"
          type="text"
          value=""
        />
        <button>
          Add Me!
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
